client/src/app/utils/utils.ts

```diff
@@ -110,17 +110,21 @@ export const getValidatedFromError = (error: unknown | undefined) => {
 export const standardURLRegex =
   /^(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s]{2,}|www\.[a-zA-Z0-9]+\.[^\s]{2,})$/;
 
-export const gitUrlRegex =
+export const gitRegularUrlRegex =
   /^(https?:\/\/[-\w.]+\/[-\w._]+\/[-\w._]+|git@[-\w.]+:[-\w._]+\/[-\w._]+)(\.git)?(\/?|#[-\d\w._]+)?$/;
 
+export const gitSshUrlRegex =
+  /^ssh:\/\/(?:[-\w.]+@)?[-\w.]+(?:(?::\d{1,5})?\/|:)[-\w._]+\/[-\w._]+(\.git)?(\/?|#[-\d\w._]+)?$/;
+
 export const standardStrictURLRegex =
   /https:\/\/(www\.)?[-a-zA-Z0-9@:%._\\+~#=]{2,256}\.[a-z]{2,4}\b([-a-zA-Z0-9@:%_\\+.~#?&//=]*)/;
 
 export const svnUrlRegex = /^svn:\/\/[^\s/$.?#].[^\s]*$/;
 
 export const customURLValidation = (schema: yup.StringSchema) => {
   const containsURL = (string: string) =>
-    gitUrlRegex.test(string) ||
+    gitRegularUrlRegex.test(string) ||
+    gitSshUrlRegex.test(string) ||
```

Comment:
That the right thing to do for sure.  I'm looking at a library that has a more comprehensive set of checks.  I'll push a change to your PR if it works nicely.